sidechain/storage/src/storage.rs

```diff
@@ -108,7 +108,8 @@ impl<SignedBlock: SignedBlockT> SidechainStorage<SignedBlock> {
 		shard_identifier: &ShardIdentifierFor<SignedBlock>,
 	) -> Result<Vec<SignedBlock>> {
 		// Ensure we find the block in storage (otherwise we would return all blocks for a specific shard).
-		if self.get_block(block_hash)?.is_none() {
+		// The exception is, if the hash is the default hash, in that case we want to return all blocks.
+		if block_hash != &BlockHash::default() && self.get_block(block_hash)?.is_none() {
```

Comment:
Minor modification for the storage interface: When the provided block hash is `Default`, we return all blocks from the storage.